recipes/pcapplusplus/all/patches/0001-Pass-CXXFLAGS-CPPFLAGS-when-compiling-3rd-party-sour.patch

```diff
@@ -0,0 +1,22 @@
+--- 3rdParty/LightPcapNg/Makefile
++++ 3rdParty/LightPcapNg/Makefile
+@@ -15,7 +15,7 @@
+ 
+ Obj/%.o: LightPcapNg/src/%.c
+ 	@echo Building file: $<
+-	@$(CC) $(INCLUDES) -Wall -O2 $(GLOBAL_FLAGS) $(DEFS) -g -c -o "$@" "$<"
++	@$(CC) $(CPPFLAGS) $(CXXFLAGS) $(INCLUDES) -Wall -O2 $(GLOBAL_FLAGS) $(DEFS) -g -c -o "$@" "$<"
+ 
+ CUR_TARGET := $(notdir $(shell pwd))
+ 
+--- 3rdParty/hash-library/Makefile
++++ 3rdParty/hash-library/Makefile
+@@ -13,7 +13,7 @@ DEFS := -DUNIVERSAL -fPIC
+ 
+ Obj/%.o: %.cpp
+ 	@echo Building file: $<
+-	@$(CXX) $(INCLUDES) -Wall -O2 $(GLOBAL_FLAGS) $(DEFS) -g -c -o "$@" "$<"
++	@$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(INCLUDES) -Wall -O2 $(GLOBAL_FLAGS) $(DEFS) -g -c -o "$@" "$<"
+ 
+ CUR_TARGET := $(notdir $(shell pwd))
```

Comment:
A better way that doesn't require patching would be adding this line in the end of `mk/platform.mk` (this file is generated automatically after running the config script):

```makefile
GLOBAL_FLAGS += $(CPPFLAGS) $(CXXFLAGS)
```

Reply:
Are you going to apply a similar patch to your upstream project?
If so, perhaps use that patch instead?

Reply:
I may do that for future versions, but for the current version `v21.05` we probably need to add it manually in the `conanfile`

Reply:
Great, let's keep it as a patch then.
Then, for a next release, we just don't have to add the patch to the patch list in `conandata.yml`.

Reply:
My original comment was that we don't need this patch...
Instead, we can run the configuration script and then add this line to the generated `mk/platform.mk` file:
```makefile
GLOBAL_FLAGS += $(CPPFLAGS) $(CXXFLAGS)
```

In future version I'll change the configuration scripts to add this line automatically

Reply:
Yes, but with a patch, no version-specific logic is added to `conanfile.py`.
As you said, a next release won't need this.

Reply:
OK, that makes sense!